File: mobile-client/ios/App/Order/OrderBusinessBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// Order-centric business bridge.
public struct OrderBusinessBridge: Sendable {
    public let order: OrderFeatureBridge
    public let shipment: OrderShipmentBridge
    public let taxation: OrderTaxationBridge

    public init(
        order: OrderFeatureBridge,
        shipment: OrderShipmentBridge,
        taxation: OrderTaxationBridge
    ) {
        self.order = order
        self.shipment = shipment
        self.taxation = taxation
    }

    public func ownedRouteMap() -> OrderOwnedRouteMap {
        OrderOwnedRouteMap()
    }
}

File: client/android/app/src/main/java/app/mobiling/client/order/OrderBusinessBridge.kt
package app.mobiling.client.order

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * Order-centric business bridge.
 *
 * Active surface is wrapper-first: Order owns Shipment / Taxation through
 * order-owned wrappers only. Legacy flat feature compatibility lives at the
 * wrapper level and no longer contaminates the active business bridge surface.
 */
class OrderBusinessBridge(
    private val order: OrderFeatureBridge,
    private val shipment: OrderShipmentBridge,
    private val taxation: OrderTaxationBridge,
) {
    fun order(): OrderFeatureBridge = order

    fun shipment(): OrderShipmentBridge = shipment

    fun taxation(): OrderTaxationBridge = taxation

    fun ownedRouteMap(): OrderOwnedRouteMap = OrderOwnedRouteMap()
}
